tests/upload.int.test.ts

```diff
@@ -17,333 +17,268 @@
-      const del = await storage.bucket(testBucket).file(name).delete();
-      return del[0].statusCode;
+
+  afterEach(async function () {
+    if (!projectID) this.skip();
+
+    const storage: Storage = this.storage;
+    const bucket = storage.bucket(this.testBucket);
+    const [files] = await bucket.getFiles();
+    const deleteOne = async (name: string): Promise<void> => {
+      await bucket.file(name).delete();
     };
-    await pMap(
-      files.map((f) => f.name),
-      uploader,
-      { concurrency: 100 },
-    );
-    const [checkFiles] = await storage.bucket(testBucket).getFiles();
-    expect(checkFiles.length).eq(0);
-  });
-  // delete bucket after all tests
-  this.afterAll(async function () {
-    if (testBucket) {
-      await storage.bucket(testBucket).delete();
-    }
-  });
 
-  it('uploads a single file', async function () {
-    const uploader = new Client();
-    const uploadResponse = await uploader.upload(testBucket, './tests/testdata/test1.txt');
-    expect(uploadResponse[0][0].name).eql('test1.txt');
-    const filesInBucket = await getFilesInBucket();
-    expect(filesInBucket.length).eq(1);
-    expect(filesInBucket).to.have.members(['test1.txt']);
+    const args: [name: string][] = files.map((file) => [file.name]);
+    await inParallel(deleteOne, args);
   });
 
-  it('uploads a single file with prefix no gzip', async function () {
-    const uploader = new Client();
-    const uploadResponse = await uploader.upload(
-      `${testBucket}/${EXAMPLE_PREFIX}`,
-      './tests/testdata/test1.txt',
-      '',
-      false,
-    );
-    const expectedFile = `${EXAMPLE_PREFIX}/test1.txt`;
-    expect(uploadResponse[0][0].name).eql(expectedFile);
-    const filesInBucket = await getFilesInBucket();
-    expect(filesInBucket.length).eq(1);
-    expect(filesInBucket).to.have.members([expectedFile]);
+  after(async function () {
+    if (!projectID) return;
+
+    const storage: Storage = this.storage;
+    const bucket = storage.bucket(this.testBucket);
+    await bucket.delete();
   });
 
-  it('uploads a single file with metadata', async function () {
-    const uploader = new Client();
-    const uploadResponse = await uploader.upload(
-      testBucket,
-      './tests/testdata/test1.txt',
-      '',
-      true,
-      true,
-      true,
-      undefined,
-      100,
-      {
-        contentType: 'application/json',
-        metadata: {
-          foo: 'bar',
-        },
-      },
-    );
-    expect(uploadResponse[0][0].name).eql('test1.txt');
-    const filesInBucket = await getFilesInBucket();
-    expect(filesInBucket.length).eq(1);
-    expect(filesInBucket).to.have.members(['test1.txt']);
-    const metadata = uploadResponse[0][0].metadata;
-    expect(metadata.contentType).eql('application/json');
-    expect(Object.keys(metadata.metadata).length).eq(1);
-    expect(metadata.metadata.foo).eql('bar');
+  it('throws an error on a non-existent bucket', async function () {
```

Comment:
Since we are refactoring - these seem like good candidates for tabletests?

Reply:
Ah now I remember why. You can't use table-driven tests when relying on context from `before`, since `this` isn't present where cases are defined 😦.